prime-router/src/main/kotlin/azure/WorkflowEngine.kt

```diff
@@ -308,23 +308,35 @@ class WorkflowEngine(
             )
             val ids = tasks.map { it.reportId }
             val reportFiles = ids
-                .map { db.fetchReportFile(it, org = null, txn = txn) }
+                .mapNotNull {
+                    try {
+                        db.fetchReportFile(it, org = null, txn = txn)
+                    } catch (e: Exception) {
```

Comment:
So, if the row is not found then db.fetchReportFile returns null (see DatabaseAccess 192).  You will only get an exception if there is more than one row or something went wrong in the DB query.  Not sure this catch does what you want.

Reply:
Catch DataAccessException instead.

Reply:
The fetchReportFile function calls error() which throws an exception when the report is not found.